src/Microsoft.DotNet.Build.Tasks.Feed/CopyBlobDirectory.cs

```diff
@@ -74,55 +83,78 @@ public async Task<bool> ExecuteAsync()
+                        // trailing slash is left off of the source and target URIs.
+                        string relativeBlobPath = blob.Uri.ToString().Substring(sourceUri.Length);
+                        string specificTargetUri = GetCanonicalStorageUri(targetUri, relativeBlobPath);
+                        BlobUrlInfo specificTargetBlobUrlInfo = new BlobUrlInfo(specificTargetUri);
+                        CloudBlob targetBlob = targetContainer.GetBlobReference(specificTargetBlobUrlInfo.BlobPath);
 
-                        Log.LogMessage($"Listing blobs in {sourceUri}");
+                        Log.LogMessage($"Merging {blob.Uri.ToString()} into {targetBlob.Uri.ToString()}");
 
-                        // Get all source URI's with the blob prefix
-                        BlobContinuationToken token = null;
-                        List<IListBlobItem> sourceBlobs = new List<IListBlobItem>();
-                        do
+                        if (!Overwrite && await targetBlob.ExistsAsync())
                         {
-                            BlobResultSegment segment = await sourceContainer.ListBlobsSegmentedAsync(sourceBlobInfo.BlobPath, true,
-                                BlobListingDetails.None, null, token, new BlobRequestOptions(), null);
-                            token = segment.ContinuationToken;
-                            sourceBlobs.AddRange(segment.Results);
+                            Log.LogError($"Target blob {targetBlob.Uri.ToString()} already exists.");
                         }
-                        while (token != null);
 
-                        // Ensure the source exists
-                        if (!SkipIfMissing && sourceBlobs.Count == 0)
+                        BlobUrlInfo specificSourceBlobUrlInfo = new BlobUrlInfo(blob.Uri);
+                        CloudBlob sourceBlob = sourceContainer.GetBlobReference(specificSourceBlobUrlInfo.BlobPath);
+                        SharedAccessBlobPolicy policy = new SharedAccessBlobPolicy
+                        {
+                            Permissions = SharedAccessBlobPermissions.Read,
+                            SharedAccessStartTime = null,
+                            SharedAccessExpiryTime = DateTimeOffset.Now.AddMinutes(30)
+                        };
+                        string sas = sourceBlob.GetSharedAccessSignature(policy);
+                        await targetBlob.StartCopyAsync(new Uri(blob.Uri + sas));
+
+                        TimeSpan waitRemaining = TimeSpan.FromMinutes(CopyWaitTimeout);
+                        TimeSpan waitInterval = TimeSpan.FromSeconds(30);
+                        ICloudBlob copyInProgessBlob;
+                        do
                         {
-                            Log.LogError($"No blobs found in {sourceUri}");
+                            await Task.Delay(waitInterval);
+                            waitRemaining.Subtract(waitInterval);
```

Comment:
This approach doesn't count time taken actually doing `GetBlobReferenceFromServerAsync`, does it?